implementations/rust/ockam/ockam_command/src/space/create.rs

```diff
@@ -59,3 +60,17 @@ async fn run_impl(
     delete_embedded_node(&opts.config, rpc.node_name()).await;
     Ok(())
 }
+
+fn validate_space_name(s: &str) -> Result<String, String> {
```

Comment:
The function should return either a `crate::Result` or a `anyhow::Result`. You can then handle the error from the caller and return a custom error message from there.

Then, both `validate_space_name` and `validate_project_name` will be the same, so you can also move the function to `./utils/api.rs` and rename it to something more generic like `validate_cloud_resource_name`.

Finally, since this is a stateless function, we would love to have some unit tests to make sure it will work with the cases described at https://github.com/build-trust/ockam/issues/3540